src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1699,6 +1699,35 @@ impl Suggestion {
             Self::RemoveDisambiguator => path_str.into(),
         }
     }
+
+    fn as_help_span(
+        &self,
+        path_str: &str,
+        ori_link: &str,
+        sp: rustc_span::Span,
+    ) -> (String, rustc_span::Span) {
+        // FIXME: if this is an implied shortcut link, it's bad style to suggest `@`
+        match self {
+            Self::Prefix(prefix) => {
+                let sp = if let Some(index) = ori_link.find('@') {
+                    sp.with_hi(sp.lo() + BytePos(index as u32 + 1))
+                } else {
+                    sp.shrink_to_lo()
+                };
```

Comment:
I usually end up writing these as a `match` (4 lines instead of 5), but it doesn't matter at all!
```suggestion
                let sp = match ori_link.find('@') {
                    Some(index) => sp.with_hi(sp.lo() + BytePos(index as u32 + 1)),
                    None => sp.shrink_to_lo(),
                };
```

Reply:
This is actually a very good point! Updating.